test/e2e/suites/init/command_line_flags_test.go

```diff
@@ -0,0 +1,187 @@
+)
+
+var karmadaEtcdExpectedExtraArgs = []string{
+	"--snapshot-count=5000",    // Override default parameters in the format --key=value.
+	"--heartbeat-interval=100", // Additional parameters are in the format --key=value.
+}
+
+var karmadaControllerManagerExpectedExtraArgs = []string{
+	"--v=2",          // Override default parameters in the format --key=value.
+	"--enable-pprof", // The format for additional parameters is --key.
+	"--skipped-propagating-namespaces=kube-system,default,my-ns", // Additional parameters are in the format --key=value1,value2.
+}
+
+var _ = ginkgo.Describe("Component command line flags testing", func() {
+	ginkgo.Context("Customize karmada components command line flags", func() {
+		var tempData string
+		var tempPki string
+		var tempDir string
+
+		ginkgo.BeforeEach(func() {
+			tempDir = os.TempDir()
+			tempData = filepath.Join(tempDir, KarmadaInstanceNamePrefix+rand.String(RandomStrLength))
+			tempPki = filepath.Join(tempData, "pki")
+		})
+
+		ginkgo.AfterEach(func() {
+			err := os.RemoveAll(tempDir)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred(), "Failed to remove temp dir")
+		})
+
+		ginkgo.It("Customize karmada components command line flags via karmadactl init command flags", func() {
+			// step1 Execute the Karmadactl init command.
+			ginkgo.By("Execute the command karmadactl init", func() {
+				args := []string{"init",
+					"--karmada-data", tempData,
+					"--karmada-pki", tempPki,
+					"--crds", crdsPath,
+					"--karmada-aggregated-apiserver-image", karmadaAggregatedAPIServerImage,
+					"--karmada-controller-manager-image", karmadaControllerManagerImage,
+					"--karmada-scheduler-image", karmadaSchedulerImage,
+					"--karmada-webhook-image", karmadaWebhookImage,
+
+					// Here we select two specific examples: etcd in StatefulSet and karmada-controller-manager in Deployment.
+					// StatefulSet etcd
+					KarmadaEtcdFlag, strings.Join(karmadaEtcdExpectedExtraArgs, ","),
+
+					// Deployment karmada-controller-manager
+					KarmadaControllerManagerFlag, strings.Join(karmadaControllerManagerExpectedExtraArgs, ","),
+				}
+
+				cmd := framework.NewKarmadactlCommand(
+					kubeconfig,
+					"",
+					karmadactlPath,
+					testNamespace,
+					KarmadactlInitTimeOut,
+					args...,
+				)
+				_, err := cmd.ExecOrDie()
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+
+			// step2 Waiting for components to be ready.
+			ginkgo.By("Waiting for components to be ready.", func() {
+				cmdinit.WaitAllKarmadaComponentReady(kubeClient, testNamespace)
+			})
+
+			// step3 Check Command Line Flags
+			ginkgo.By("Check command line flags", func() {
+				gomega.Eventually(func() bool {
+					klog.Infof("Checking command line flags for %s", cmdinit.KarmadaEtcdName)
+					// 1. Get StatefulSet
+					statefulSet, err := kubeClient.AppsV1().StatefulSets(testNamespace).
+						Get(context.Background(), cmdinit.KarmadaEtcdName, metav1.GetOptions{})
```

Comment:
```suggestion
						Get(context.TODO(), cmdinit.KarmadaEtcdName, metav1.GetOptions{})
```
No special reason; Just for consistency.